packages/mds-compliance-service/service/validators.ts

```diff
@@ -14,68 +14,115 @@
-  provider_id: Joi.string().uuid().required().error(Error('provider_id is missing')),
-  policy: policySchema,
-  vehicles_found: Joi.array().items(matchedVehicleInformationSchema).required(),
-  excess_vehicles_count: Joi.number().required(),
-  total_violations: Joi.number().required()
+export const { $schema: matchedVehicleInformationSchema } = SchemaValidator<MatchedVehicleInformation>({
+  $id: 'MatchedVehicleInformation',
+  type: 'object',
+  properties: {
+    device_id: uuidSchema,
+    rule_applied: uuidSchema,
+    rules_matched: { type: 'array', items: uuidSchema },
+    state: vehicleStatusSchema,
+    event_types: { type: 'array', items: vehicleEventTypeSchema },
+    timestamp: timestampSchema,
+    gps: {
+      type: 'object',
+      properties: {
+        lat: { type: 'number', minimum: -90, maximum: 90 },
+        lng: { type: 'number', minimum: -180, maximum: 180 }
+      },
+      required: ['lat', 'lng']
+    },
+    speed: { type: 'number' }
+  },
+  required: ['event_types', 'state', 'timestamp']
 })
 
-export const ValidateComplianceSnapshotDomainModel = (
-  complianceSnapshot: ComplianceSnapshotDomainModel
-): ComplianceSnapshotDomainModel => {
-  const { error } = complianceSnapshotDomainModelSchema.validate(complianceSnapshot)
-  if (error) {
-    throw new ValidationError(error.message, complianceSnapshot)
-  }
-  return complianceSnapshot
-}
-
-const getComplianceSnapshotsByTimeIntervalOptionsSchema = Joi.object().keys({
-  start_time: Joi.number().integer().required().error(Error('start_time not provided')).less(Joi.ref('end_time')),
-  end_time: Joi.number().integer().allow(null),
-  provider_ids: Joi.array().items(uuidSchema).allow(null),
-  policy_ids: Joi.array().items(uuidSchema).allow(null)
+export const { validate: validateComplianceSnapshotDomainModel } = SchemaValidator<ComplianceSnapshotDomainModel>({
+  $id: 'ComplianceSnapshotDomainModel',
+  type: 'object',
+  properties: {
+    compliance_snapshot_id: uuidSchema,
+    compliance_as_of: timestampSchema,
+    provider_id: uuidSchema,
+    policy: {
+      $id: 'Policy',
+      type: 'object',
+      properties: { policy_id: uuidSchema, name: { type: 'string' } },
+      required: ['name', 'policy_id']
+    },
+    vehicles_found: { type: 'array', items: matchedVehicleInformationSchema },
+    excess_vehicles_count: { type: 'integer' },
+    total_violations: { type: 'integer' }
+  },
+  required: [
+    'compliance_as_of',
+    'compliance_snapshot_id',
+    'excess_vehicles_count',
+    'policy',
+    'provider_id',
+    'total_violations',
+    'vehicles_found'
+  ]
 })
 
-export const ValidateGetComplianceSnapshotsByTimeIntervalOptions = (
-  options: GetComplianceSnapshotsByTimeIntervalOptions
-) => {
-  if (!isDefined(options.end_time)) {
-    // eslint-disable-next-line no-param-reassign
-    options.end_time = now()
-  }
+export const { validate: validateGetComplianceSnapshotsByTimeIntervalOptions } =
```

Comment:
It looks like in the original version of this, `end_time` defaulted to `now()`, and `start_time` defaulted to `end_time`. Here, it seems like `start_time` defaults to `now` (through the usage of the dynamicDefaults keyword), but `end_time` doesn't seem to default (but I may be missing something).